pycloudlib/openstack/instance.py

```diff
@@ -156,3 +164,40 @@ def wait_for_delete(self):
     def wait_for_stop(self):
         """Wait for instance stop."""
         self.conn.compute.wait_for_server(self.server, status='SHUTOFF')
+
+    def add_network_interface(self) -> str:
+        """Add nic to running instance.
+
+        Returns IP address in string form
+        """
+        port = self.conn.network.create_port(
+            network_id=self.network_id,
+            node_id=self.server.id
+        )
+        self.added_local_ports.append(port.id)
+        interface = self.conn.compute.create_server_interface(
+            server=self.server.id,
+            port_id=port.id
+        )
+        return interface['fixed_ips'][0]['ip_address']
+
+    def _get_port_id_by_ip(self, ip_address: str):
+        ports = self.conn.network.ports()
+        for port in ports:
+            for ip in port['fixed_ips']:
+                if ip['ip_address'] == ip_address:
+                    return port
+        raise Exception('Could not find port with IP: {}'.format(ip_address))
+
+    def remove_network_interface(self, ip_address: str):
+        """Remove nic from running instance."""
+        port = self._get_port_id_by_ip(ip_address)
```

Comment:
We aren't handling the potential Exception here if invalid IP is provided. Don't we want to handle this Exception by logging a warning as below or just let it break the CI run?

Reply:
I don't think so. If we're trying to remove a NIC and fail, I think we have a problem that needs to be dealt with. I used a warning further down because that's our own accounting of the ports and likely just a harmless bug on our end if it's not there.